macros/stitch/base/stitch_fb_ad_creatives.sql

```diff
@@ -8,39 +8,94 @@
 {% macro default__stitch_fb_ad_creatives() %}
 
 with base as (
+    
+    select * from {{ var('ad_creatives_table') }}
+    
+),
+
+child_links as (
+    
+    select * from {{ ref('fb_ad_creatives__child_links') }}
+    
+),
+
+{% if target.type == 'snowflake' %}
+
+links_joined as (
```

Comment:
@axelazaid we can't do this! :) you're in the "default" version of this macro--if you need to do something specific to snowflake, you need to do that using a snowflake version of the macro, just like we do everywhere else. We can never write an `if` statement like this, it's very bad for maintainability of this code moving forwards.

I'm happy to talk this through with you, it's something i've also discussed with erin and chris recently as well.

Reply:
i think i understand where you're coming from, but i'm happy to go through this with you!

Reply:
@axelazaid this PR is getting a bit stale at this point. Is this still something that you'd like to get merged? at this point, the if/then logic is the blocker here. this would need to be rewritten in order to get the PR merged.

Reply:
@jthandy i made those changes last week in this branch. there's still some changes that need to be done on the url-parsing aspect though (related to the most recent pr of coalescing `url_tags` & `url`), before this could be looked at agaiin